### PR TITLE
A number of improvements to the grid handling

### PR DESCRIPTION
**Pull request by busstoptaktik**
Two interpolation tests in ntv2/mod.rs (succesfully validated against ESRI's ntv2_cvt program)

Improve check for NTv2 file header sanity

Improve readability and correct vector capacity in ntv2 parse_subgrid_grid()

Reduce visibility of SubGridHeader

Clean up the `use` block in ntv2 subgrid.rs

In the inverse case, don't iterate in reverse order over the grid collection in deformation & gridshift operators (we still want to hit the same grid as in the forward case)

### CODE REVIEW

**Comment by busstoptaktik** [comment]
@Rennzie does this look OK to you?

**Comment by Rennzie** [comment]
🤩 Looks great @busstoptaktik !

**Comment by busstoptaktik** [comment]
> I'm curious to know if the precision is about what you'd expect given the off by half differences you'd noticed between the PROJ interpolation that of RG?

@Rennzie no, these differences are probably too large, and I'm staring myself to blindness today, checking interpolation code between ESRI's `ntv2_cvt`, PROJ's `hgridshift` and Rust Geodesy's `gridshift`.

PROJ and ESRI agree on the tenth to hundredth of a millimeter, while RG is sightly off, only agreeing with the other two on the millimeter level.

I have even reimplemented ESRI's code in RG, with next to no difference, so I assume the problem is in the calculation of the cell relative coordinates.

I'm looking into it, but it is a nasty problem, not the least because the grid is only single precision, so where and when you convert to f64 in the process will also make a difference.

My comparison routine is this:

```console
$ echo 2.1686 41.3874 0 0 | cct -d 5 proj=pipeline step proj=hgridshift grids=./path/to/100800401.gsb step proj=utm zone=31 --

> 430391.39938  4581973.06191

$ ntv2_cvt -f .\path\to\100800401.gsb  41.3874 2.1686 | proj -f %9.5f +proj=utm +zone=31

> 430391.39938    4581973.06190

$ echo  41.3874 2.1686 | cargo run -- "geo:in | gridshift grids=100800401.gsb | utm zone=31"

> 430391.39952 4581973.06152
```

In summary:
```txt
PROJ: 430391.39938  4581973.06191
ESRI: 430391.39938  4581973.06190
RG:   430391.39952  4581973.06152
```

**Comment by Rennzie** [comment]
Thanks for looking into it. Please let me know if I can help in any way.
